go-controller/pkg/networkmanager/network_controller.go

```diff
@@ -342,6 +342,10 @@ func (c *networkController) ensureNetwork(network util.MutableNetInfo) error {
 
 	err = nc.Start(context.Background())
 	if err != nil {
+		nc.Stop()
+		if err := nc.Cleanup(); err != nil {
+			klog.Errorf("Failed to cleanup network %s: %v", networkName, err)
+		}
```

Comment:
This is what I would do:

- I would not call `Stop` here. It might work nicely now but semantically an implementation might need to tell apart if its being told to stop after successfully starting vs after unsuccessfully starting. The implementation is already aware about the latter case so `Stop` can be reserved just for the former case. `Cleanup` is fine since there is no semantic contract that something needs to be started in order to be cleaned up.
- I would add network state with `stoppedAndDeleting=true` if cleanup failed to retry cleanup and block it from starting until that has happened.